### PR TITLE
Reporter query

### PR DESCRIPTION
**Pull request by colmazia**
## Implementation details

add reporters query for validator to query their granted reporters via grpc query

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: x/oracle/keeper/grpc_query.go [hunk 1]
```diff
@@ -190,23 +191,30 @@ func (k Querier) IsReporter(c context.Context, req *types.QueryIsReporterRequest
...
+	granterGrantsRequest := &authz.QueryGranterGrantsRequest{
+		Granter: granter,
+	}
+	granterGrantsRes, err := k.authzKeeper.GranterGrants(c, granterGrantsRequest)
+	if err != nil {
+		return nil, err
+	}
+	reporters := make([]string, 0)
+	for _, rep := range granterGrantsRes.Grants {
+		if rep.Authorization.GetCachedValue().(authz.Authorization).MsgTypeURL() == sdk.MsgTypeURL(&types.MsgReportData{}) {
```
Comment (RogerKSI) [inline x/oracle/keeper/grpc_query.go:213]: Please check the expiration and add test cases for an expired grants.

**Review by RogerKSI — CHANGES_REQUESTED** [review]

File: x/oracle/keeper/grpc_query.go (hunk 1 above)
Comment (colmazia) [reply x/oracle/keeper/grpc_query.go]: Done

**Review by RogerKSI — APPROVED** [review]

File: x/oracle/keeper/grpc_query.go [hunk 2]
```diff
@@ -190,23 +191,31 @@ func (k Querier) IsReporter(c context.Context, req *types.QueryIsReporterRequest
...
-	// reporters := make([]string, len(reps))
-	// for idx, rep := range reps {
-	// 	reporters[idx] = rep.String()
-	// }
-	return &types.QueryReportersResponse{Reporter: []string{}}, nil
+	val, err := sdk.ValAddressFromBech32(req.ValidatorAddress)
+	if err != nil {
+		return nil, err
+	}
+	granter := (sdk.AccAddress(val)).String()
```
Comment (taobun) [inline x/oracle/keeper/grpc_query.go:204]: Remove parenthesis -> `sdk.AccAddress(val).String()`

File: x/oracle/keeper/grpc_query.go [hunk 3]
```diff
@@ -190,23 +191,31 @@ func (k Querier) IsReporter(c context.Context, req *types.QueryIsReporterRequest
...
+	granterGrantsRequest := &authz.QueryGranterGrantsRequest{
+		Granter: granter,
+	}
+	granterGrantsRes, err := k.authzKeeper.GranterGrants(c, granterGrantsRequest)
+	if err != nil {
+		return nil, err
+	}
+	reporters := make([]string, 0)
+	for _, rep := range granterGrantsRes.Grants {
+		if (rep.Authorization.GetCachedValue().(authz.Authorization).MsgTypeURL() == sdk.MsgTypeURL(&types.MsgReportData{})) && rep.Expiration.After(ctx.BlockTime()) {
```
Comment (taobun) [inline x/oracle/keeper/grpc_query.go:214]: Remove first `(`

File: x/oracle/types/expected_keepers.go [hunk 4]
```diff
@@ -1,6 +1,7 @@
 package types
 
 import (
+	context "context"
```
Comment (taobun) [inline x/oracle/types/expected_keepers.go:4]: weird

**Review by taobun — CHANGES_REQUESTED** [review]

File: x/oracle/keeper/grpc_query.go (hunk 2 above)
Comment (colmazia) [reply x/oracle/keeper/grpc_query.go]: Done

File: x/oracle/keeper/grpc_query.go (hunk 3 above)
Comment (colmazia) [reply x/oracle/keeper/grpc_query.go]: Done

File: x/oracle/types/expected_keepers.go (hunk 4 above)
Comment (colmazia) [reply x/oracle/types/expected_keepers.go]: Fixed

**Review by taobun — APPROVED** [review]

**Pull request merged**